tslearn/metrics/cysax.py

```diff
@@ -0,0 +1,150 @@
+STUFF_cysax = "cysax"
+
+import numpy as np
+from numba import njit, prange
+from sklearn.linear_model import LinearRegression
+
+__author__ = "Romain Tavenard romain.tavenard[at]univ-rennes2.fr"
+
+DTYPE = float
+DTYPE_INT = int
+
+
+@njit(parallel=True)
+def inv_transform_paa(dataset_paa, original_size):
+    n_ts = dataset_paa.shape[0]
+    sz = dataset_paa.shape[1]
+    d = dataset_paa.shape[2]
+    i = 0
+    t = 0
+    di = 0
+    t0 = 0
+    seg_sz = original_size // sz
+    dataset_out = np.zeros((n_ts, original_size, d))
+
+    for i in prange(n_ts):
+        for t in range(sz):
+            t0 = t * seg_sz
+            for di in range(d):
+                dataset_out[i, t0 : t0 + seg_sz, di] = dataset_paa[i, t, di]
+    return dataset_out
+
+
+@njit(parallel=True)
+def cydist_sax(sax1, sax2, breakpoints, original_size):
+    assert sax1.shape[0] == sax2.shape[0] and sax1.shape[1] == sax2.shape[1]
```

Comment:
How about `assert sax1.shape == sax2.shape`?

Reply:
Are you sure that both matrices sax1 and sax2 are of dimension 2?

Reply:
Maybe that I can write: `assert sax1.shape[:2] == sax2.shape[:2]`
However, according to the name of the variables, the matrices sax1 and sax2 should be of dimension 2

Reply:
Ahh..I just assumed :) so, maybe we can do .shape[:2] if it works(?)

Reply:
According to the previous cython file cysax.pyx, the inputs sax1 and sax2 should be of dimension 2.

Reply:
> According to the previous cython file cysax.pyx, the inputs sax1 and sax2 should be of dimension 2.

I think regardless of their `ndim`, we can replace `sax1.shape[0] == sax2.shape[0] and sax1.shape[1] == sax2.shape[1]` . with `sax1.shape[:2] = sax2.shape[:2]`. Aren't they the same? what do you think?